composer/core/callback.py

```diff
@@ -58,246 +46,29 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
         try:
             event_cb = getattr(self, event.value)
```

Comment:
I'm not really sure under what circumstances this would actually succeed. We'd need something like:

```python
class MyCallback(Callback):
    def batch_start(...):
        super().run_event(...)
        ...
```
Do you think it's fine to just drop the deprecation warning?

Reply:
I'd like to keep it in for now in case if there are any user-implemented callbacks that are no longer getting triggered.

Reply:
The engine calls run_event, so if it's not overridden, then this check would happen (and deprecation warnings would be raised)